### ISSUE TITLE
complete freeze when using search and replace 

### ISSUE THREAD

**Issue by danielhrisca** [spyder-ide/spyder#14086]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

Ever since Spyder version 4 I often get complete freezes when I use the search and replace function. For this reason I have downgraded to version 3 

I think it is related to pasting to the search and replace input fields but it doesn't happen all the time.

### What steps reproduce the problem?

1. CTRL+R
2. try to paste in the input fields

### What is the expected output? What do you see instead?

I have to kill Spyder using the task manager.

### Paste Traceback/Error Below (if applicable)

```python-traceback

No traceback available

```

## Versions

* Spyder version: 4.1.5
* Python version: 3.7.4 64bit
* Qt version: 5.12.8
* PyQt version: 5.12.3
* Operating System name/version: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.4.1 (OK)
diff_match_patch >=20181111    :  20181111 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.14.0 (OK)
jedi =0.17.1                   :  0.17.1 (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  0.9.2 (OK)
paramiko >=2.4.0               :  2.7.1 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.0 (OK)
pygments >=2.0                 :  2.6.1 (OK)
pylint >=1.0                   :  2.5.2 (OK)
pyls >=0.34.0;<1.0.0           :  0.35.1 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.7.2 (OK)
qtconsole >=4.6.0              :  4.7.3 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
sphinx >=0.6.6                 :  3.0.3 (OK)
spyder_kernels >=1.9.4;<1.10.0 :  1.9.4 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  19.0.1 (OK)

# Optional:
cython >=0.21                  :  0.29.17 (OK)
matplotlib >=2.0.0             :  3.2.1 (OK)
numpy >=1.7                    :  1.19.2 (OK)
pandas >=0.13.1                :  1.1.2 (OK)
scipy >=0.17.0                 :  1.4.1 (OK)
sympy >=0.7.3                  :  1.5.1 (OK)

```

**Comment by dalthviz** [comment]
Hi @danielhrisca sorry to hear that, I'm not totally sure but maybe the error happening is a segmentation fault. Checking your dependencies to setup an env to test seems like you have installed qt `5.12.8`. Could you try to update qt to `5.12.9`?

Any new info in order to replicate this issue is greatly appreciated :)

**Comment by danielhrisca** [comment]
I will install the PyQt version, but like I said I have this problem ever since verion 4.0.0 with multiple PyQt versions

**Comment by dalthviz** [comment]
Checking a little bit maybe this is related with #12662

Probably what is happening is that the replace operation takes to long to be finished and then if you try to do something else while the replace is taking place Spyder freezes. Its the behavior described in #12662 similar to your case?

Maybe you can upload a GIF for us to understand better what you are experiencing and find a way to replicate the issue in a more consistent manner. Without a way to replicate it is kind of difficult to see what we can do, sorry :/

**Comment by danielhrisca** [comment]
The problem does not come suddenly. After some time of using the IDE I notice a certain lag when performing search and replace, and after that the freeze comes (before starting this issue I let the IDE for ~10minutes and it did not recover, so I had to kill the process). I will try to catch the behavior

**Comment by dalthviz** [comment]
Thank you for the new info @danielhrisca and any new info regarding this is greatly appreciated :)

**Comment by danielhrisca** [comment]
@dalthviz 

I could not replicate the freeze lately. 

For large files I saw that the replace command takes a long time to complete, so it is possible that this happened in the past and I did not have the patience to wait. During this time the GUI is completely unresponsive.

Any reason why this action should take so much time?

**Comment by dalthviz** [comment]
@danielhrisca my initial guess is that the replace operation is being done over the GUI thread making Spyder unresponsive while the replace is being done. The replace itself taking a long time to finish maybe could be caused by the Editor trying to update any GUI related mark like warnings in each replace but not totally sure if that is actually the cause. @ccordoba12 do you have any idea why this is happening?

Also, taking into account that this issue is related with the performance of the find replace functionality maybe we should close this one in favor of #12662 @ccordoba12 ?

**Comment by danielhrisca** [comment]
The operation is really slow even with lining disabled

**Comment by dalthviz** [comment]
Maybe this is related with #14176

**Comment by danielhrisca** [comment]
> Maybe this is related with #14176

probably

**Comment by ccordoba12** [comment]
Closing as a duplicate of #14176 then.
